lib/ledgerBlockStorage.js

```diff
@@ -26,55 +26,58 @@ module.exports = class LedgerBlockStorage {
 
   /**
    * Adds a block in the ledger given a block, metadata associated with the
-   * block, and a set of options.
+   * block
    *
    * @param block - the block to create in the ledger.
    * @param meta - the metadata associated with the block.
    *   blockHash - the hash value of the block.
-   * @param options - a set of options used when creating the block.
    * @param callback(err) - the callback to call when finished.
    *   err - An Error if an error occurred, null otherwise.
    *   result - the result of the operation.
    *     block - the block that was committed to storage.
    *     meta - the metadata that was committed to storage.
    */
-  add(block, meta, options, callback) {
-    if(typeof options === 'function') {
-      callback = options;
-      options = {};
-    }
+  add({block, meta}, callback) {
     if(!meta.blockHash) {
       return callback(new BedrockError(
         'A block hash for the given block was not specified.',
-        'BadRequest', {meta}));
+        'DataError', {meta}));
```

Comment:
`DataError` seems vague. Seems like we should be throwing a `TypeError` for misuse of the API (or passing it to the callback). Thoughts?

Reply:
Yes, I think a TypeError is appropriate and I will update all the other APIs as well.